src/ucp/wireup/select.c

```diff
@@ -383,22 +387,28 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
      * best one has the highest score (from the dedicated score_func) and
      * has a reachable tl on the remote peer */
     UCS_BITMAP_FOR_EACH_BIT(tl_bitmap, rsc_index) {
-        resource   = &context->tl_rscs[rsc_index].tl_rsc;
-        iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
-        md_attr    = &context->tl_mds[context->tl_rscs[rsc_index].md_index].attr;
+        resource_desc  = &context->tl_rscs[rsc_index];
+        resource       = &resource_desc->tl_rsc;
+        iface_attr     = ucp_worker_iface_get_attr(worker, rsc_index);
+        md_attr        = &context->tl_mds[resource_desc->md_index].attr;
+        local_md_flags = criteria->local_md_flags;
 
-        if ((context->tl_rscs[rsc_index].flags & UCP_TL_RSC_FLAG_AUX) &&
+        if ((resource_desc->flags & UCP_TL_RSC_FLAG_AUX) &&
             !(criteria->tl_rsc_flags & UCP_TL_RSC_FLAG_AUX)) {
             continue;
         }
 
         if (select_params->ep_init_flags & UCP_EP_INIT_CONNECT_TO_IFACE_ONLY) {
             local_iface_flags |= UCT_IFACE_FLAG_CONNECT_TO_IFACE;
+        } else if (ucp_wireup_connect_p2p(worker, rsc_index, has_cm)) {
+            /* Remove INVALIDATE md flag in case of transport uses CONNECT_TO_EP
+             * connection mode */
```

Comment:
instead of removing the flag don't add it in first place

Reply:
we can't do it, since we don't know whether a concrete transport support INVALIDATE or not

Reply:
ok (i guess you mean we don't know if it's p2p or not)

Reply:
/* We should not need MD invalidate support for p2p lanes, since both sides close the connection in case of error */

Reply:
done